feat: Allow tweaking catalog name in Trino (#903)
stackabletech/trino-operator: docs/modules/trino/pages/usage-guide/catalogs/index.adoc

A `TrinoCluster` can, once created, detect and use new catalogs that have been subsequently created with a matching label. This also means that it is possible to reuse a `TrinoCatalog` within multiple `TrinoClusters`.

=== Catalog name tweaking

By default the name of the catalog in Trino is inferred from the `.metadata.name` of the TrinoCatalog object.

> This ensures that no catalog names clash, as their can only be one TrinoCatalog with a given name.

Reviewer: ```suggestion
This ensures that no catalog names clash, as there can only be one TrinoCatalog with a given name.
```

One inconvenience is that you need to quote catalogs (or schemas and tables for that matter) containing `-` in Trino, while `_` is fine.

> As Kubernetes doesn't allow `_` in the object names, we offer a feature to replace `-` with `_`, which allows you to use valid Kubernetes names, but keeps the convenience of `_` in catalog names.

Reviewer: ```suggestion
As Kubernetes doesn't allow `_` in the object names, we offer a feature to replace `-` with `_`, which allows you to use valid Kubernetes names, but keeps the convenience of using `_` in catalog names.
```

> You can enable it like this, the Trino catalog will be called `my_postgres`:

Reviewer: ```suggestion
In order to replace `-` with `_`, use the setting `name.inferred.replaceHyphensWithUnderscores`.
The Trino catalog will be called `my_postgres`:
```
